src/doctor.ts

```diff
@@ -1,13 +1,22 @@
-export interface DoctorResult {
+type NonEmptyArray<A> = { 0: A } & A[];
+
+export type DoctorResult = {
   /** The name of Doctor */
   title: string;
   /** Head text to be displayed always */
   headerText: string;
-  /** Suggestion Doctor messages that contain suggestions if no targets are found */
-  messages?: DoctorMessage[];
-  /** Status information for each target */
-  targets?: DoctorTargetInfo[];
-}
+} & (
+  | {
+      /** Suggestion Doctor messages that contain suggestions if no targets are found */
+      messages: NonEmptyArray<DoctorMessage>;
+      targets?: never;
+    }
+  | {
+      messages: never;
+      /** Status information for each target */
+      targets?: NonEmptyArray<DoctorTargetInfo>;
+    }
+);
```

Comment:
This better represents what Metals actually sends: if `messages` is present, then it's non-empty and `targets` is missing. Conversely, if `targets` is present, then it's non-empty and `messages` is missing.